### DISCUSSION TITLE
Need help giving access to Docker in QNAP container station

### DISCUSSION THREAD

**Discussion by CrownParkComputing** [coder/coder#6571] [General] [answered]
I have my Coder container running, when I try to start a template I get :
Error: Error pinging Docker server: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/_ping": dial unix /var/run/docker.sock: connect: permission denied

I have looked at the docker apps to try and configure but not sure if needed, If Portainer can create docker containers and yacht do I need some extra permission added ??
Thanks

**Comment by matifali** [comment, marked as answer]
If you are running a Coder as a docker container you need to mount the docker daemon socket as volume so that Coder can create containers on the host. 

```shell
export CODER_DATA=$HOME/.config/coderv2-docker
export DOCKER_GROUP=$(getent group docker | cut -d: -f3)
mkdir -p $CODER_DATA
docker run --rm -it \
    -v $CODER_DATA:/home/coder/.config \
    -v /var/run/docker.sock:/var/run/docker.sock \
    --group-add $DOCKER_GROUP \
    ghcr.io/coder/coder:latest

```
See https://coder.com/docs/v2/latest/platforms/docker for more information.
